backend/src/main/scala/cromwell/backend/WorkflowBackendActor.scala

```diff
@@ -29,37 +31,68 @@ object WorkflowBackendActor {
+  case object InitializationSuccess extends InitializationResponse
+  case class InitializationFailed(reason: Throwable) extends InitializationResponse
 }
 
 /**
   * Workflow-level actor for executing, recovering and aborting jobs.
   */
 trait WorkflowBackendActor extends Actor with ActorLogging {
 
+  /**
+    * The execution context for the actor
+    */
   protected implicit def ec: ExecutionContext
+
+  /**
+    * The set of calls which this backend will be expected to run
+    */
+  protected def calls: Seq[Call]
+
+  /**
+    * The workflow descriptor for the workflow in which this Backend is being used
+    */
   protected def workflowDescriptor: BackendWorkflowDescriptor
+
+  /**
+    * The configuration for the backend, in the context of the entire Cromwell configuration file.
+    */
   protected def configurationDescriptor: BackendConfigurationDescriptor
 
   def receive: Receive = LoggingReceive {
-    case Execute(jobDescriptor) => performActionThenRespond(execute, jobDescriptor, onFailure = descriptorToExecutionFailure)
-    case Recover(jobDescriptor) => performActionThenRespond(recover, jobDescriptor, onFailure = descriptorToExecutionFailure)
-    case Abort(jobKey)          => performActionThenRespond(abort, jobKey, onFailure = AbortFailed)
+    case Initialize             => performActionThenRespond(initSequence, onFailure = InitializationFailed)
+    case Execute(jobDescriptor) => performActionThenRespond( () => execute(jobDescriptor), onFailure = executionFailed(jobDescriptor.key))
+    case Recover(jobDescriptor) => performActionThenRespond( () => recover(jobDescriptor), onFailure = executionFailed(jobDescriptor.key))
+    case Abort(jobKey)          => performActionThenRespond( () => abort(jobKey), onFailure = abortFailed(jobKey))
+  }
+
+  override def postStop = {
+    afterAll()
   }
 
-  private def performActionThenRespond[A](operation: A => Future[WorkflowBackendActorResponse],
-                                          operationArgument: A,
-                                          onFailure: (A, Throwable) => WorkflowBackendActorResponse) = {
-    // To avoid sending a response to the wrong `sender` if more than one `execute` arrives before the first one finishes,
-    // for example:
-    val sndr = sender()
-    operation(operationArgument) onComplete {
-      case Success(r) => sndr ! r
-      case Failure(t) => sndr ! onFailure(operationArgument, t)
+  private def performActionThenRespond(operation: () => Future[WorkflowBackendActorResponse],
+                                       onFailure: (Throwable) => WorkflowBackendActorResponse) = {
+    val respondTo: ActorRef = sender
+    operation.apply onComplete {
+      case Success(r) => respondTo ! r
+      case Failure(t) => respondTo ! onFailure(t)
     }
   }
 
+  /**
+    * Our predefined sequence to run during preStart
+    */
+  final def initSequence() = for {
+    _ <- beforeAll()
+    result <- validate()
+  } yield result
```

Comment:
Perhaps I'm missing some stuff, but this is making me wonder:

1. Why does validation happen after beforeAll, which does things like upload auth files that really shouldn't be uploaded if the workflow doesn't validate?
2. Why is this implemented with `Future`s?  This `WorkflowActor` is open for business once `preStart` returns, but if we've spun off a `Future` we're not waiting to see if these prerequisite operations have completed.
3. It feels weird that we dictate to an implementer that they must write a `beforeAll` and then that they must have it call this particular method.